Added Traccar tracker support

ToRvaLDz opened the pull request:
**Description:**
Adds support for a new device tracker based on Traccar GPS Tracking Platform (https://www.traccar.org/).
Traccar supports person presence detection based on GPS device.

This tracker uses the Traccar API to determine the device location.

**Example entry for configuration.yaml:**
```YML
device_tracker:
  - platform: traccar
    url: <http://server:port>
    username: <username>
    password: <password>
```

ToRvaLDz committed:
Added Traccar tracker support

homeassistant commented:
Hi @ToRvaLDz, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#15160).

Once you do that we will be able to review and accept this pull request.

Thanks!

frenck commented:
Added `docs-missing`, since I cannot find a matching PR on the documentation repo.

ToRvaLDz committed:
Code cleanup

ToRvaLDz committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into dev

ToRvaLDz committed:
Removed unused variables

ToRvaLDz committed:
Merge with upstream

frenck commented:
Docs PR has been opened, removing the `docs-missing` label.

frenck commented:
@ToRvaLDz You might wanna check the code you've actually committed this PR.
There are all kinds of files in here, that should not have been in this PR.

For example:
- <https://github.com/home-assistant/home-assistant/pull/15160/files#diff-830968569b50f87c82e22a0042a7f760>
- <https://github.com/home-assistant/home-assistant/pull/15160/files#diff-d9ae695c8119e58c2ec435b6deae2d53>

ToRvaLDz committed:
Revert "Merge with upstream"

This reverts commit 82c86c397aaf949c53fe8b27b1bf65fcc33abc80.

fabaff commented on homeassistant/components/device_tracker/traccar.py:
```diff
@@ -0,0 +1,117 @@
+            response = requests.get(
+                url, auth=(self.username, self.password), timeout=4)
+        except requests.exceptions.Timeout:
+            """ Wrong URL or server down. """
```
Misplaced docstring.

fabaff commented on homeassistant/components/device_tracker/traccar.py:
```diff
@@ -0,0 +1,117 @@
+        if response.status_code == 200:
+            return response.json()
+        elif response.status_code == 401:
+            """ Authentication error. """
```
Dito.

fabaff commented on homeassistant/components/device_tracker/traccar.py:
```diff
@@ -0,0 +1,117 @@
+
+
+class TraccarScanner(object):
+
```
Docstring is missing.

fabaff commented on homeassistant/components/device_tracker/traccar.py:
```diff
@@ -0,0 +1,117 @@
+        self.see = see
+        self.url = config.get(CONF_URL)
+        self.username = config.get(CONF_USERNAME)
+        self.password = config.get(CONF_PASSWORD)
```
You should check the credentials during the setup of the platform and make the setup fails.

ToRvaLDz committed:
Fixed Misplaced docstring
Added docstring in TraccarScanner class
Added check for credentials during platform setup

ToRvaLDz committed:
Fixed Misplaced docstring

MartinHjelmare reviewed (changes requested):
I think the requests and parsing should be moved to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

kjetilpp commented:
Any updates? Cannot wait anymore 🙈

olenoerby commented:
What's the progress? It's really interesting!
